socialhome/streams/app/stores/streamStore.state.js

```diff
@@ -1,24 +1,14 @@
-import _isNumber from "lodash/isNumber"
 import _get from "lodash/get"
+import _keyBy from "lodash/keyBy"
 
 
 export default function () {
+    const streamName = _get(window, ["context", "streamName"], "")
     return {
-        translations: {
-            stampedContent: {
-                h2: _get(window, ["context", "translations", "stampedContent", "h2"], ""),
-                p: _get(window, ["context", "translations", "stampedContent", "p"], ""),
-            },
-        },
-        contentList: _get(window, ["context", "contentList"], []),
-        stream: {
-            hasNewContent: false,
-            newContentLengh: 0,
-            streamName: _get(window, ["context", "streamName"], ""),
-        },
-        isUserAuthenticated: _get(window, ["context", "isUserAuthenticated"], false),
-        showAuthorBar: _get(window, ["context", "showAuthorBar"], false),
-        currentBrowsingProfileId: (_isNumber(_get(window, ["context", "currentBrowsingProfileId"], undefined))
-            ? `${window.context.currentBrowsingProfileId}` : undefined),
+        contentList: _keyBy(_get(window, ["context", "contentList"], []), item => item.id),
```

Comment:
Btw, I hope this doesn't mess with the order since the stream must be rendered in the correct order, not by id.

Reply:
> Btw, I hope this doesn't mess with the order since the stream must be rendered in the correct order, not by id.

Good question. Object are not ordered in JS. So my implementation is wrong. I have to use [Maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

Reply:
Ah, crap. Maps won't work either. Vue can only watch native objects.

Reply:
What about storing the list and the objects separately, like;

    contentList: [1,2,3],
    contents: {1: {}, 2:{}, 3:{}},

This way 1) order is guaranteed and 2) objects can be found easily from the components, ensuring we don't need to duplicate store into components as mutable local state.

This allows also inserting directly to the `contentList` from a websocket `new` message. When the object itself appears in `contents`, it will then render it.